File: leetcode/1/1.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        numsMap = {}
        for indx, v in enumerate(nums):
            complement = target - v
            if complement in numsMap:
                return [numsMap[complement], indx]
            numsMap[v] = indx
        return []


sol = Solution()
nums = [2,7,11,15]
target = 9
res = sol.twoSum(nums, target)
print(res)

File: leetcode/1/1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let numsMap = {};
    for(let i = 0; i<nums.length; i++){
        complement = target - nums[i];
        if (complement in numsMap){
            return [numsMap[complement], i]
        }
        numsMap[nums[i]] = i
    }
    return [];
};

console.log(twoSum([2, 7, 11, 15], 9)); 